File: Python/selection_sort.py
#selection_sort function

def selection_sort(arr):
    for i in range(0,len(arr)-1):
        min_index=i;
        for j in range(i+1,len(arr)):
            if arr[j] < arr[min_index]:
                min_index=j;
        arr[i],arr[min_index]=arr[min_index],arr[i]


            
arr=[2,5,3,7,9,0,-5,-1]
print("Before Sorting")

for i in range(0,len(arr)):
    print(arr[i],end=" ")
  
  
selection_sort(arr)
print("\n")
print("After Sorting")

for i in range(0,len(arr)):
    print(arr[i],end=" ")  
    

File: C and C++/selection_sort.cpp
#include<iostream>
using namespace std;


void swap(int *a,int *b){
    int temp;
    temp=*a;
    *a=*b;
    *b=temp;

}


void selection_sort(int *arr,int n){

    // 10,4,5,0,44

    for(int i=0;i<n-1;i++){
        int min_index=i;
        for(int j=i+1;j<n;j++){

                //only chnage inequality sign for increasing order
            if(arr[j] > arr[min_index] )
                    min_index=j;
            



        }
        swap(&arr[i],&arr[min_index]);


    }



}

void print_array(int arr[],int n){
    cout<<endl;
    for(int i=0;i<n;i++){
        cout<<arr[i]<<" ";
    }
    cout<<endl;
}
int main(){


    int arr[]={10,4,5,0,44,-5,-2};
    int n=sizeof(arr)/sizeof(arr[0]);
    printf("Before Sorting");
    print_array(arr,n);
    printf("After SortingSorting");
    selection_sort(arr,n);
    print_array(arr,n);



}




